x-pack/plugins/fleet/server/services/epm/packages/get.ts

```diff
@@ -142,6 +150,90 @@ export async function getPackages(
   return packageListWithoutStatus;
 }
 
+export async function getInstalledPackages(options: {
+  savedObjectsClient: SavedObjectsClientContract;
+  type?: PackageDataStreamTypes;
+  nameQuery?: string;
+  pageAfter?: SortResults;
+  pageSize: number;
+  sortDirection: 'asc' | 'desc';
+}) {
+  const { savedObjectsClient, pageAfter, pageSize, nameQuery, sortDirection, type } = options;
+
+  const packageSavedObjects = await savedObjectsClient.find<Installation>({
+    type: PACKAGES_SAVED_OBJECT_TYPE,
+    // Pagination
+    perPage: pageSize,
+    ...(pageAfter && { searchAfter: pageAfter }),
+    // Sort
+    sortField: 'name',
+    sortOrder: sortDirection,
+    // Name filter
+    ...(nameQuery && { searchFields: ['name'] }),
+    ...(nameQuery && { search: `${nameQuery}* | ${nameQuery}` }),
+    filter: nodeBuilder.and([
+      // Filter to installed packages only
+      nodeBuilder.is(
+        `${PACKAGES_SAVED_OBJECT_TYPE}.attributes.install_status`,
+        installationStatuses.Installed
+      ),
+      // Filter for a "queryable" marker
+      buildFunctionNode(
+        'nested',
+        `${PACKAGES_SAVED_OBJECT_TYPE}.attributes.installed_es`,
+        nodeBuilder.is('type', 'index_template')
+      ),
+      // "Type" filter
+      ...(type
+        ? [
+            buildFunctionNode(
+              'nested',
+              `${PACKAGES_SAVED_OBJECT_TYPE}.attributes.installed_es`,
```

Comment:
We use `installed_es` here as other properties within the SO aren't mapped, and therefore aren't queryable. We need something top level so that we can take advantage of pagination on the Saved Objects.